packages/x-date-pickers/src/DesktopNextDatePicker/DesktopNextDatePicker.test.tsx

```diff
@@ -92,22 +92,71 @@ describe('<DesktopNextDatePicker />', () => {
     });
   });
 
-  it('switches between views uncontrolled', () => {
-    const handleViewChange = spy();
-    render(
-      <DesktopNextDatePicker
-        open
-        componentsProps={{ toolbar: { hidden: false } }}
-        defaultValue={adapterToUse.date(new Date(2018, 0, 1))}
-        onViewChange={handleViewChange}
-      />,
-    );
-
-    fireEvent.click(screen.getByLabelText(/switch to year view/i));
-
-    expect(handleViewChange.callCount).to.equal(1);
-    expect(screen.queryByLabelText(/switch to year view/i)).to.equal(null);
-    expect(screen.getByLabelText('year view is open, switch to calendar view')).toBeVisible();
+  describe('Views', () => {
+    it('should switch between views uncontrolled', () => {
+      const handleViewChange = spy();
+      render(
+        <DesktopNextDatePicker
+          open
+          componentsProps={{ toolbar: { hidden: false } }}
+          defaultValue={adapterToUse.date(new Date(2018, 0, 1))}
+          onViewChange={handleViewChange}
+        />,
+      );
+
+      fireEvent.click(screen.getByLabelText(/switch to year view/i));
+      expect(handleViewChange.callCount).to.equal(1);
+      expect(screen.queryByLabelText(/switch to year view/i)).to.equal(null);
+      expect(screen.getByLabelText('year view is open, switch to calendar view')).toBeVisible();
+    });
+
+    it('should go to the first view when re-opening the picker', () => {
+      const handleViewChange = spy();
+      render(
+        <DesktopNextDatePicker
+          defaultValue={adapterToUse.date(new Date(2018, 0, 1))}
+          onViewChange={handleViewChange}
+          slotProps={{ toolbar: { hidden: false } }}
+        />,
+      );
+
+      openPicker({ type: 'date', variant: 'desktop' });
+
+      fireEvent.click(screen.getByLabelText(/switch to year view/i));
+      expect(handleViewChange.callCount).to.equal(1);
+
+      // Dismiss the picker
+      userEvent.keyPress(document.activeElement!, { key: 'Escape' });
+
+      openPicker({ type: 'date', variant: 'desktop' });
+      expect(handleViewChange.callCount).to.equal(2);
```

Comment:
I do not have a full understanding of how the current view is managed. But this test raised a question:

Would it makes sense to switch back to the default view when we close the view. Such that when reopening, the current view will not be set and so we avoid getting two renderings:
- one for opening
- a second one for updating the `view` state

I'm wondering if removing the early return `if(!open)` in hook you modified would do the job

Reply:
> Would it makes sense to switch back to the default view when we close the view.

That's an interesting question indeed
The view would probably not be reset if the user is controlling the opening state and closing it from the outside.
Not sure if this is problematic.